### PR TITLE
Stop custom handling network down error

### PR DESCRIPTION
**Pull request by bgavrilMS**
…Essentials for that.

Fixes bug https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/592 by providing a **consistent** approach to HTTP errors. See more details in the bug.

Note that this is theoretically a breaking change because we are changing exception logic, but in practice it should not make a difference, because if the network is down, MSAL will fail with an Http exception when making a discovery call.

### CODE REVIEW

File: src/Microsoft.Identity.Client/MsalError.cs [hunk 1]
```diff
@@ -470,6 +470,7 @@ public static class MsalError
         /// <para>Mitigation [App development]</para> In the application you could either inform the user that there are network issues
         /// or retry later
         /// </summary>
+        [Obsolete("MSAL no longer throws this error - it will allow the HttpClient exceptions to propagate. App developers may write their own logic for detecting access to the network issues, for example by using Xamarin.Essentials. ")]
```
Comment (MarkZuber) [inline src/Microsoft.Identity.Client/MsalError.cs]: Is this change related to a bug/issue around this area?  Curious what prompted the change.

Comment (jennyf19) [reply src/Microsoft.Identity.Client/MsalError.cs]: @MarkZuber I think this ->[592](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/592)

**Comment by henrik-me** [comment]
@jmprieur : Do you consider this a breaking change that would require a major version bump? I'm generally ok with this change as long as we provide guidance on this in the wiki and to be considered to include handling in samples.
